lib/src/create_report.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:pana/pana.dart';
+import 'package:path/path.dart' as p;
+import 'package:source_span/source_span.dart';
+import 'package:yaml/yaml.dart';
+
+import '../models.dart';
+import 'pubspec.dart';
+
+Future<Report> createReport(
+    String packageDir, ToolEnvironment toolEnvironment) async {
+  Pubspec pubspec;
+  try {
+    pubspec = Pubspec.parseFromDir(packageDir);
+  } on Exception catch (e) {
+    throw Exception('Cannot not create report without a pubspec.yaml: $e');
+  }
+
+  return Report(sections: [
+    _supportsDart2(pubspec),
+    // TODO(sigurdm):Implement rest of sections.
+  ]);
+}
+
+ReportSection _supportsDart2(Pubspec pubspec) {
+  final supportsDart2 = pubspec.supportsDart2;
+  final issues = <_Issue>[];
+
+  if (!supportsDart2) {
+    final environment = pubspec.toJson()['environment'] as Map;
```

Comment:
We should probably expose `pubspec._inner.environment` instead.

Reply:
Done